monoio/src/driver/op/read.rs

```diff
@@ -18,58 +15,103 @@ use crate::{
     BufResult,
 };
 
+macro_rules! read_result {
+    ($($name:ident<$T:ident : $Trait:ident> { $buf:ident }),* $(,)?) => {
+        $(
+            impl<$T: $Trait> super::Op<$name<$T>> {
+                pub(crate) async fn result(self) -> BufResult<usize, $T> {
+                    let complete = self.await;
+
+                    // Convert the operation result to `usize`
+                    let res = complete.meta.result.map(|v| v as usize);
+                    // Recover the buffer
+                    let mut buf = complete.data.$buf;
+                    let read_len = *res.as_ref().unwrap_or(&0);
```

Comment:
Here's a difference comparing to current impl, if op failed, buf will be set_init with 0.
I think it's safer than current impl. But I don't know if it is compatiable well.

Reply:
We should investigate if the kernel can guarntee it returns positive `n` when it has touched the buffer.

Reply:
I think we can leave it alone.
Possibility 1: The kernel can guarantee that the buffer will not be touched when returning -1, so there is no need for us to do so, and the user can assume the legitimacy of the internal data.
Possibility 2: The kernel may have touched the buffer when returning -1, and whether it has been written depends on the type of error returned, which I think can be left to the user. Unconditionally set_init(0) is not reasonable, and in some cases the user can continue to use the last data in the buffer. This is not a memory safety issue, but a data validity issue.

Reply:
you're right.